alf/algorithms/handcrafted_algorithm.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) 2021 Horizon Robotics and ALF Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Handcrafted Algorithm."""
+
+import numpy as np
+import gin
+import torch
+
+import alf
+from alf.algorithms.config import TrainerConfig
+from alf.algorithms.off_policy_algorithm import OffPolicyAlgorithm
+from alf.data_structures import AlgStep, Experience, LossInfo, TimeStep
+from alf.tensor_specs import BoundedTensorSpec
+
+
+@gin.configurable
+class HandcraftedAlgorithm(OffPolicyAlgorithm):
+    """An Algorithm with handcrafted computational logic.
```

Comment:
Comment that a concrete algorithm should subclass from this and implement _policy_func

Reply:
Comments added